build/solr/scripts/solr-healthcheck.pl

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/perl
```

Comment:
Please avoid introducing Perl in repositories that don’t use it already.

Python is also available in Solr image.

Reply:
I only chose Perl because Python wasn't available in the image for me:

```sh
solr@22a1a3f26448:/opt/solr-9.7.0$ python
bash: python: command not found
solr@22a1a3f26448:/opt/solr-9.7.0$ perl --version

This is perl 5, version 34, subversion 0 (v5.34.0) built for x86_64-linux-gnu-thread-multi
[...]
```

Am I missing something? I'm fine with installing Python in the image instead. I could even use bash + jq. But I figured a small amount of Perl would be fine for a tiny one-off script.

Reply:
Oops, my bad, I mistook with another image and didn’t actually checked Solr image.

Yes, Bash would be fine.

To detect whether Solr is running locally or listening to the outside, I didn’t find anything useful in the APIs. This seems to work:
`ps auxww | grep Djetty.host=localhost`

Reply:
I tried using lsof as discussed on IRC, and although it easily worked for detecting whether Solr was available for outside connections, the solr-exporter container still wasn't happy until the collections were healthy:
```
solr-exporter  | INFO  - 2025-01-31 17:33:40.329; org.apache.solr.client.solrj.impl.CloudSolrClient; Request to collection [annotation] failed due to (510) org.apache.solr.common.SolrException: Could not find a healthy node to handle the request., retry=0 maxRetries=5 commError=false errorCode=510 
```
Merging the lsof check with the collection health check script seems to work well, though.